add: mining medic (ready)

NightDawnFox opened the pull request:
<!-- Пишите **НИЖЕ** заголовков и **ВЫШЕ** комментариев, иначе ваш текст может не отобразиться. -->
<!-- В Contributing.MD вы можете найти некоторые рекомендации к оформлению пулл-реквеста. -->

## Описание
Добавляет новую игровую роль - шахтёрского врача. 
Множество новых специфичных спрайтов, модификация карт под новую раундстарт роль, ремаппинг аванпостов лаваленда под новый отдел, несколько новых, уникальных для шахтера предметов.
Дополнительно, офицерские камеры.

<!-- Опишите, что делает ваш Pull request. Документировать каждую деталь не требуется, просто укажите основные изменения. -->

## Причина создания ПР / Почему это хорошо для игры
https://discord.com/channels/617003227182792704/755125334097133628/1293990774375186635 - шахтерский медик
https://discord.com/channels/617003227182792704/755125334097133628/1353256128171212891 - нагрудные камеры для офицеров
https://discord.com/channels/617003227182792704/755125334097133628/1353256234915991574 - вырубание планшетов при ионной буре

<!-- Здесь можно оставить ссылку на сообщение в #отчеты-по-предложениям, чтобы подтвердить, что предложение обсуждалось внутри Discord-сообщества. -->
<!-- Если отчёта нет, то укажите, почему это изменение положительно влияет на игру. -->
<!-- В случае исправления бага, укажите ссылку на канал в #баг-репорты-v2 или issue в репозитории. В ином случае, опишите баг и ступени для его воспроизведения. -->
<!-- Пример ссылки в Discord-сообщество : https://discord.com/channels/617003227182792704/755125334097133628/ID-сообщения -->

## Демонстрация изменений
Всё в дискорде
<!-- В случае наличия изменений, влияющих на игровую часть, опишите их здесь. В случае их отсутствия, этот пункт можно удалить -->

## Тесты
зашел, проверил манифесте. Новая ролька спавнится нормально, новые вещи для нее выглядят хорошо, карты изменены и работают хорошо
<!-- Здесь необходимо описать шаги, которые предпринимались для тестирования изменения. Этот пункт обязателен, без него Pull request будет рассматриваться дольше. -->

NightDawnFox committed:
1

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into proba_pera_2-electric_boogalo

NightDawnFox committed:
шахтерский монитор

NightDawnFox committed:
спрайты + фиксы + тгуи

NightDawnFox committed:
карточка

NightDawnFox committed:
перевод

NightDawnFox committed:
блядь

NightDawnFox committed:
перевод + джампсьюты

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into proba_pera_2-electric_boogalo

NightDawnFox committed:
перевод + 4 новые шмотки

У КОСТЮМА НЕТ СПРАЙТОВ, кисик доделывает

NightDawnFox committed:
Update jobs.dm

NightDawnFox committed:
лабхалат + перевод

NightDawnFox committed:
пиздаа

NightDawnFox committed:
ой.

Rafnik02 commented on code/game/objects/items/devices/sensor_device.dm:
```diff
@@ -85,3 +85,23 @@
+	name = "mining crew monitor"
+	desc = "Миниатюрное устройство, с помощью которого можно отслеживать датчики членов экипажа станции. Эта модель настроена на шахтерский персонал станции."
+	ru_names = list(
+		NOMINATIVE = "шахтерский монитор экипажа",
```
шахтЕрский и прочее ниже 🤓

Rafnik02 commented on code/modules/clothing/suits/labcoat.dm:
```diff
@@ -24,47 +32,125 @@
+	name = "mining medic's labcoat"
+	desc = "Стерильный белый халат с коричневыми нашивками на плечиках. На нём видны следы пепла."
+	ru_names = list(
+		NOMINATIVE = "лабораторный халат шахтерского врача",
```
```suggestion
		NOMINATIVE = "лабораторный халат шахтёрского врача",
```
🤓

Rafnik02 commented on code/modules/clothing/under/jobs/plasmamen/medsci.dm:
```diff
@@ -1,76 +1,179 @@
+	name = "mining medic plasma envirosuit"
+	desc = "Защитный костюм плазмолюда, созданный специально для шахтерских врачей."
+	ru_names = list(
+		NOMINATIVE = "защитный костюм шахтерского врача",
```
```suggestion
		NOMINATIVE = "защитный костюм шахтёрского врача",
```

PlayerUnknown14 commented:
Уэээ

Rafnik02 commented on code/modules/mining/equipment/mining_medic.dm:
```diff
@@ -0,0 +1,118 @@
+	name = "mining camera"
+	desc = "Небольшая нагрудная видеокамера, обладающая массивным датчиком, позволяющим считывать датчики костюма с основной станции."
+	ru_names = list(
+		NOMINATIVE = "шахтерская видеокамера",
```
```suggestion
		NOMINATIVE = "шахтёрская видеокамера",
```

Rafnik02 commented on code/modules/mining/equipment/mining_medic.dm:
```diff
@@ -0,0 +1,118 @@
+
+/obj/item/clothing/accessory/mining_camera/examine(mob/user)
+	. = ..()
+	. += span_notice("Данный тип камер позволяет вести трансляцию как на планшет шахтерского медика, так и в развлекательную сеть станции. На текущий момент камера [on ? "в" : "вы"]ключена.")
```
```suggestion
	. += span_notice("Данный тип камер позволяет вести трансляцию как на планшет шахтёрского медика, так и в развлекательную сеть станции. На текущий момент камера [on ? "в" : "вы"]ключена.")
```

Rafnik02 commented on code/modules/mining/equipment/mining_medic.dm:
```diff
@@ -0,0 +1,118 @@
+	ru_names = list(
+		NOMINATIVE = "шахтерский монитор видеокамер",
+		GENITIVE = "шахтерского монитора видеокамер",
+		DATIVE = "шахтерскму монитору видеокамер",
```
```suggestion
		DATIVE = "шахтёрскому монитору видеокамер",
```

Rafnik02 reviewed (commented):
насрал ревьювом

PlayerUnknown14 committed:
Шахтер -> шахтёр

PlayerUnknown14 commented on code/game/objects/items/devices/radio/headset.dm:
```diff
@@ -158,229 +206,501 @@
 /obj/item/radio/headset/heads/magistrate
 	name = "magistrate's headset"
-	desc = "The headset of the Magistrate."
+	desc = "Эта гарнитура принадлежит тому, кого на этой станции ни во что не ставят. По крайней мере до тех пор, пока офицерам не требуется приказ на казнь."
```
Грубо...

PlayerUnknown14 committed:
Merge branch 'master220' into proba_pera_2-electric_boogalo

NightDawnFox committed:
рекомпайл тгуи

PlayerUnknown14 committed:
Правки локализации

NightDawnFox commented on code/modules/clothing/head/jobs.dm:
```diff
@@ -200,7 +204,7 @@
 /obj/item/clothing/head/hooded/monk_hood
 	name = "monk hood"
-	desc = "Капюшон, покрывающий голову. Деревянная доска покупается отдельно."
+	desc = "Капюшон, покрывающий голову. Деревянный посох покупается отдельно."
```
я думал шутка в досках, которые они разбивают руками..

PlayerUnknown14 replied:
А у меня наоборот, монахи ассоциируются с посохами. Лады, делай как хочешь.

NightDawnFox replied:
да и так и так можно, все равно

NightDawnFox commented on code/modules/clothing/suits/labcoat.dm:
```diff
@@ -46,7 +46,7 @@
 /obj/item/clothing/suit/storage/labcoat/mad
 	name = "mad scientist's labcoat"
-	desc = "Ношение этого халата побуждает в вас желание врезать кому-нибудь по башке и выбросить его тело в шлюз."
+	desc = "Ношение этого халата побуждает в вас желание врезать кому-нибудь по башке и выбросить его тело в космос."
```
https://neolurk.org/wiki/Выбросить_в_шлюз

PlayerUnknown14 replied:
Будто бы одно и то же.

NightDawnFox replied:
вполне

NightDawnFox committed:
маппинг

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into proba_pera_2-electric_boogalo

NightDawnFox committed:
рекомпайл тгуи

NightDawnFox committed:
123

NightDawnFox committed:
ой.

NightDawnFox committed:
карты 2

NightDawnFox committed:
Update mining_medic.dm

NightDawnFox committed:
пиздец

NightDawnFox committed:
карты 2: electric boogalo

NightDawnFox committed:
фикс

NightDawnFox committed:
фикс 2

NightDawnFox committed:
да еб твою мать.

NightDawnFox committed:
ладно

Antoonij commented on code/modules/mining/equipment/mining_medic.dm:
```diff
@@ -0,0 +1,149 @@
+	else
+		news_feed = !news_feed
+		balloon_alert(user, "ретрансляция [news_feed ? "в" : "вы"]ключена!")
+		return CLICK_ACTION_SUCCESS
```
Здесь можно без else, все равно ретарним

Antoonij commented on code/modules/mining/equipment/mining_medic.dm:
```diff
@@ -0,0 +1,149 @@
+	ru_names = list(
+		NOMINATIVE = "шахтёрский монитор видеокамер",
+		GENITIVE = "шахтёрского монитора видеокамер",
+		DATIVE = "шахтёрскму монитору видеокамер",
```
```suggestion
		DATIVE = "шахтёрскому монитору видеокамер",
```

Antoonij commented on code/modules/mining/equipment/mining_medic.dm:
```diff
@@ -0,0 +1,149 @@
+	icon_state = "mining_monitor"
+	lefthand_file = 'icons/mob/inhands/lavaland/misc_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/lavaland/misc_righthand.dmi'
+	item_state	= "mining_monitor"
```
```suggestion
	item_state = "mining_monitor"
```
стиль съехал

Antoonij commented on code/modules/mining/equipment/mining_medic.dm:
```diff
@@ -0,0 +1,149 @@
+
+/obj/item/clothing/accessory/mining_camera/update_icon_state()
+	. = ..()
+	icon_state = "mining_camera[on ? "_on" : ""]"
```
вместо mining_camera лучше использовать initial(icon_state), позволит легче расширять код

Antoonij commented on code/modules/mining/equipment/mining_medic.dm:
```diff
@@ -0,0 +1,149 @@
+
+/obj/item/clothing/accessory/mining_camera/Destroy()
+	if(on)
+		update_camera_state()
```
Сомнительно вызывать это тут, ибо этот прок:

ожидает юзера, ибо должен вывестись тучат - будет рантайм, ибо тут его не передаем
имеются лишние действия, в виде обновления иконки и состояния, что только замедляет удаление объекта

NightDawnFox committed:
Update suits.dmi

NightDawnFox committed:
блин

PlayerUnknown14 committed:
Правки локализации

PlayerUnknown14 committed:
И ещё микроправка

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into proba_pera_2-electric_boogalo

NightDawnFox committed:
Update tgui.bundle.js

NightDawnFox committed:
фиксы

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into proba_pera_2-electric_boogalo

NightDawnFox committed:
Update tgui.bundle.js

Antoonij commented on code/modules/mining/equipment/mining_medic.dm:
```diff
@@ -0,0 +1,214 @@
+
+/obj/item/clothing/accessory/camera/Destroy()
+	if(on)
+		if(news_feed)
```
эти чеки в дестрое не нужны, нам нужно в любом случае уже несуществующий объект убрать отовсюду и почистить ссылки

Antoonij commented on code/modules/tgui/modules/crew_monitor.dm:
```diff
@@ -63,4 +63,6 @@
 		if(CREW_VISION_SECURITY)
 			data["isBP"] = 1
+		if(CREW_VISION_MINING)
+			data["isMM"] = 1
```
```suggestion
			data["isMM"] = TRUE
```
это в джсе уже юзается как бул, обозначим и тут

Antoonij commented on code/__DEFINES/job.dm:
```diff
@@ -49,12 +49,13 @@
 #define JOB_FLAG_MIME				(1<<12)
 #define JOB_FLAG_CIVILIAN			(1<<13)
 #define JOB_FLAG_EXPLORER			(1<<14)
+#define JOB_FLAG_MINING_MEDIC 		(1<<15)
```
тут стиль съехал, немного правее этот битфлаг

NightDawnFox committed:
Update code/modules/tgui/modules/crew_monitor.dm

Co-authored-by: Antoonij <42318445+Antoonij@users.noreply.github.com>

NightDawnFox committed:
Update job.dm

NightDawnFox committed:
Merge branch 'proba_pera_2-electric_boogalo' of https://github.com/NightDawnFox/Paradise into proba_pera_2-electric_boogalo

NightDawnFox committed:
Update mining_medic.dm

NightDawnFox committed:
срочнофиксы

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into proba_pera_2-electric_boogalo

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into proba_pera_2-electric_boogalo

NightDawnFox committed:
Update tgui.bundle.js

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into proba_pera_2-electric_boogalo

NightDawnFox committed:
да

NightDawnFox committed:
фиксы проблемных спрайтов

NightDawnFox committed:
фикс улучшения брони майнинг медика

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into proba_pera_2-electric_boogalo

NightDawnFox committed:
боже как же я люблю рекомпиллить тгуи сукааа

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into proba_pera_2-electric_boogalo

NightDawnFox committed:
Update tgui.bundle.js

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into proba_pera_2-electric_boogalo

NightDawnFox committed:
Update tgui.bundle.js

ПРИШЛО ВРЕМЯ ПЕРЕКОМПИЛЛИРИРОВАТЬ ТГУИ ТГУИ САМ СЕБЯ НЕ ПЕРЕКОМПИЛЛИРУЕТ ПЕРЕКОМПИЛЛИРУЙ ЕГО, ПЕРЕКОМПИЛЛИРУЙ ЕГО ЕЩЕ РАЗ, ЗАЧЕМ МНЕ ХТМЛ, У МЕНЯ НЕТ ВРЕМЕНИ ЧТОБЫ ЕБАТЬСЯ С НИМ ЛУЧШЕ ЕЩЕ РАЗ ПЕРЕКОМПИЛЛИРОВАТЬ ТГУИ.
Я ПЕРЕКОМПИЛЛИРУЮ ТГУИ 20 РАЗ В ДЕНЬ. КАЖДАЯ РЕКОМПИЛЛЯЦИЯ ЗАНИМАЕТ ПЯТЬ МИНУТ. Я ЖИВУ АКТИВНОЙ И ПОЛНОЦЕННОЙ ЖИЗНЬЮ, Я УСПЕШЕН А ПОЭТОМУ ТРОГАЮ ДЖАВАСКРИПТ, А ПОСЛЕ ЭТОГО ПЕРЕКОМПИЛЛИРУЮ ТГУИ
ТУПЫЕ АДЕПТЫ ХТМЛ ОДЕРЖИМЫ РИСОВАНИЕМ НЕКРОСИМВОЛОВ, А Я СВОБОДНЫХ ОТ ЗАДРОТСТВА ЧЕЛОВЕК НОВЫЙ ПР С ТГУИ ЗАЛИЛИ РАНЬШЕ МОЕГО ГИТХАБ БОТ АКТИОНС АДДЕД ЗЕ МЕРЖ КОНФЛИКТ ЛАБЕЛ, ЛУЧШЕ Я ПЕРЕКОМПИЛЛИРУЮ ТГУИ ЕЩЕ РАЗ И ЗАЛЬЮ, СТАБИЛЬНОСТЬ НЕ НУЖНА
Я НЕ ПЕРЕКОМПИЛЛИРОВАЛ ТГУИ НЕДЕЛЮ, ПОЙДУ ПЕРЕКОМПИЛЛИРУЮ, В ТГУИ ЖЕ ВСЕ ПРОСТО И ПОНЯТНО
ААААААААА ЗИС БРАНЧ ХАС КОНФЛИКТС ЗЭТ МАСТ БИ РЕЗОЛВЕД, ЭТО ЖЕ ОЧЕВИДНО КАК ЕЁ РЕШИТЬ. ПРИШЛО ВРЕМЯ ПЕРЕКОМПИЛЛИРОВАТЬ ТГУИ
КОКОКОКОКОКОКО ХТМУЛЬ, УДОБСТВО, ПИТУХИ КОКОКОКОКОКОКО
